src/snyk/cli/services/cliService.ts

```diff
@@ -26,32 +32,46 @@ export abstract class CliService<CliResult> extends AnalysisStatusProvider {
     this.analysisStarted();
     this.beforeTest();
 
-    const process = new CliProcess(this.logger, this.config);
+    const cliPath = CliExecutable.getPath(this.extensionContext.extensionPath);
+    const checksumCorrect = await this.isChecksumCorrect(cliPath);
+    if (!checksumCorrect) {
+      const error = new CliError('Snyk CLI is corrupt. Please reinstall the extension.');
```

Comment:
Why ask user to reinstall the extention?
Shell we try to redownload/update CLI first?

Reply:
Makes sense, good point 👍 I wanted to cover the most basic behaviour here, hence no additional logic implemented. Added it now :)